config/http_config_test.go

```diff
@@ -347,14 +378,61 @@ func TestNewClientFromConfig(t *testing.T) {
 					InsecureSkipVerify: false},
 			},
 			handler: func(w http.ResponseWriter, r *http.Request) {
-				username, password, ok := r.BasicAuth()
-				if !ok {
-					fmt.Fprintf(w, "The Authorization header wasn't set")
-				} else if ExpectedUsername != username {
-					fmt.Fprintf(w, "The expected username (%s) differs from the obtained username (%s).", ExpectedUsername, username)
-				} else if ExpectedPassword != password {
-					fmt.Fprintf(w, "The expected password (%s) differs from the obtained password (%s).", ExpectedPassword, password)
+				// Example server response header:
+				// WWW-Authenticate:  Digest realm="prometheus", nonce="43568ca162f46c3bcc57ecae193b3159", qop="auth", opaque="3bc9f19d8195721e24469ff255750f8c", algorithm=MD5, stale=FALSE
+				//
+				// Example client request header:
+				// Authorization: Digest username="foo", realm="prometheus", nonce="43568ca162f46c3bcc57ecae193b3159", uri="/", cnonce="NDA2M2JmYzQ2YTQ4OTQ0OTQ1NzE0NmI3ZmYyY2YyNzU=", nc=00000001, qop=auth, response="fe543d7eeb2d2f0aba8d100a1f076909", opaque="3bc9f19d8195721e24469ff255750f8c", algorithm=MD5
+
+				const (
+					nonce = "43568ca162f46c3bcc57ecae193b3159"
+					realm = "prometheus"
+				)
+
+				if authHeader := r.Header.Get("Authorization"); authHeader == "" {
+					// first request
+					w.Header().Set("www-authenticate", "Digest realm=\""+realm+"\", nonce=\""+nonce+"\", qop=\"auth\", opaque=\"3bc9f19d8195721e24469ff255750f8c\", algorithm=MD5, stale=FALSE")
+					w.WriteHeader(401)
```

Comment:
can we refactor this test to allow varying the inputs?
There's a couple of cases I think this needs to be able to test:
- qop=auth-int
- different algorithms
- algorithm=(alg)-sess vs algorithm=(alg) (all the way to include cnonces)

Later on, if we can keep some state: nextnonce support

Reply:
Thanks for your review! I could include additional tests for different combinations of these values if requested. But I'm not sure if that's the way to go. The digest auth client library already includes tests for these (e.g. see [here](https://github.com/icholy/digest/blob/1fc1d2b310fbf1d9f743482810b3c0b43ebc4f80/transport_test.go#L14)). While we surely want to test that digest auth works in _general_, I don't think it's our job to test the inner working of the library again.